postgresql/provider.go

```diff
@@ -40,6 +41,12 @@ func Provider() *schema.Provider {
 				}, false),
 			},
 			"host": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				DefaultFunc: schema.EnvDefaultFunc("PGHOST", nil),
+				Description: "Name of PostgreSQL server address",
+			},
+			"connection_host": {
```

Comment:
I am also in need of this feature 😄

As a potential user, I think I would find the stated difference between `host` and `connection_host` a bit confusing. As far as I know, the problem we're trying to solve is specific to the RDS IAM auth. I think it'd be more clear if the new provider option was prefixed with `aws_rds_iam_` like the other options below (i.e. `aws_rds_iam_hostname`). Then we could clearly document that it should be the internal hostname of the RDS instance, used to generate an IAM token, which might not be the same address used to connect to the instance if connecting over a VPN / tunnel

Reply:
Thanks for the reasonable comments! I am thinking about how I should update this PR. 

IAM specific host is going to be the default host name though, and the connection host is the host through VPN/Tunnel. Not everyone uses VPN/Tunnel, but mostly every uses the host to get IAM token. 

If I add a new IAM one, when not specified, we will use the default host to get the token, and when specified, we will use the default host to connect (VPN/Tunnel one), and use iam one to get the token. Sounds like a plan?

Reply:
updated!